common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -39,17 +86,735 @@ class NativeAuthMsalController : BaseNativeAuthController() {
+                    )
+                }
+                is SignInTokenApiResult.CodeIncorrect -> {
+                    SignInCommandResult.IncorrectCode(
+                        error = tokenApiResult.error,
+                        errorDescription = tokenApiResult.errorDescription,
+                        errorCodes = tokenApiResult.errorCodes
+                    )
+                }
+
+                is SignInTokenApiResult.UnknownError, is SignInTokenApiResult.InvalidAuthenticationType,
+                is SignInTokenApiResult.MFARequired, is SignInTokenApiResult.InvalidCredentials,
+                is SignInTokenApiResult.UserNotFound -> {
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result: $tokenApiResult"
+                    )
+                    tokenApiResult as ApiErrorResult
+                    ICommandResult.UnknownError(
+                        error = tokenApiResult.error,
+                        errorDescription = tokenApiResult.errorDescription,
+                        details = tokenApiResult.details,
+                        errorCodes = tokenApiResult.errorCodes
+                    )
+                }
+            }
+        } catch (e: Exception) {
+            Logger.error(TAG, "Exception thrown in signInSubmitCode", e)
+            throw e
+        }
+    }
+
+    /**
+     * Makes a call to the resend code endpoint, mapping responses returned from the server into a command result.
+     */
+    fun signInResendCode(parameters: SignInResendCodeCommandParameters): SignInResendCodeCommandResult {
+        LogSession.logMethodCall(TAG, "${TAG}.signInResendCode")
+
+        try {
+            val oAuth2Strategy = createOAuth2Strategy(parameters)
+
+            val result = performSignInChallengeCall(
+                oAuth2Strategy = oAuth2Strategy,
+                credentialToken = parameters.credentialToken
+            )
+            return when (result) {
+                is SignInChallengeApiResult.OOBRequired -> {
+                    SignInCommandResult.CodeRequired(
+                        credentialToken = result.credentialToken,
+                        codeLength = result.codeLength,
+                        challengeTargetLabel = result.challengeTargetLabel,
+                        challengeChannel = result.challengeChannel,
+                    )
+                }
+                is SignInChallengeApiResult.PasswordRequired -> {
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result: $result"
+                    )
+                    ICommandResult.UnknownError(
+                        error = "unexpected_api_result",
+                        errorDescription = "API returned unexpected result: $result"
+                    )
+                }
+                SignInChallengeApiResult.Redirect -> {
+                    ICommandResult.Redirect()
```

Comment:
Did we conclude where the correlation ID should be comming from @SammyO @shahzaibj ? If we should be getting one from ESTS, we should input it here, rather than using the default DiagnosticContext value

Reply:
I gave feedback on the previous PR to grab it from eSTS.

Reply:
Thanks folks. We're working on a ticket right now to fix this across the board, including a review of how our native auth APIs work. Let's leave this as is; we'll be fixing this in a future version. 
note: correlation IDs implementation are a bit of a placeholder - we're aware they're not reliable.

Reply:
@SammyO Can you share the link to the PBI here? (and also add a TODO in the code along with link to PBI to fix this)

Reply:
https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2710164

Reply:
@SaurabhMSFT  can you update this?